src/tests/ftest/pool/dynamic_server_pool.py

```diff
@@ -78,42 +66,56 @@ def verify_uuids(self):
         self.expected_uuids.sort()
         self.assertEqual(self.expected_uuids, actual_uuids)
 
+    def check_pool_location(self, hosts, uuid_to_ranks):
+        """Iterate pools and verify that the UUID-named directory exists in
+        each hosts.
+
+        Args:
+            hosts (list of str): Hostnames to search the UUID-named directory.
+            uuid_to_ranks (str to list of int dictionary): UUID to rank list
+                dictionary.
+        """
+        RC_SUCCESS = 0
+        for pool in self.pool:
+            # Note that we don't check mapping between rank and hostname, but it
+            # appears that self.hostlist_servers[0] is always rank0, 1 is rank1,
+            # and the extra server we'll be adding will be rank2.
+            for rank, host in enumerate(hosts):
+                rc = check_for_pool(host, pool.uuid.lower())
+                pool_exists_on_host = rc == RC_SUCCESS
+                # If this rank is in the rank list, there should be the
+                # UUID-named directory; i.e., pool_exist_on_host is True.
+                self.assertEqual(
+                    rank in uuid_to_ranks[pool.uuid.lower()],
+                    pool_exists_on_host)
+
     def test_dynamic_server_pool(self):
         """
-        JIRA ID: DAOS-3598
+        JIRA ID: DAOS-3595
 
         Test Description: See class description.
 
         :avocado: tags=all,medium,control,full_regression,dynamic_server_pool
         """
-        self.pool = []
+        uuid_to_ranks = {}
 
         # Create a pool on rank0.
-        self.create_new_pool("0")
+        self.pool.append(self.get_pool(create=False))
+        self.pool[-1].target_list.update([0], "pool.target_list")
+        self.pool[-1].create()
+        self.expected_uuids.append(self.pool[-1].uuid.lower())
+        uuid_to_ranks[self.pool[-1].uuid.lower()] = [0]
```

Comment:
This probably should have remained a class method as the code is repeated four times - assuming the method supports passing in a rank list of None.  Adding `self.uuid_to_ranks = {}` in `__init__()` would help enable this.

If you wanted to improve the maintenance of the assumption that the `self.server_manager[0].hosts` list index directly correlates to rank number, we could add a method - maybe `ServerManager.get_ranks()` - which implements this current behavior.  This way if something changes we can modify one method for all tests as opposed to multiple tests.  Not necessarily needed, but something to think about.

Reply:
I created create_pool_with_ranks and moved these lines there.
I think it's a good idea to handle the mapping of rank-hostname in one place. I'll ask about the mapping to the control plane people.